TakingITGlobal/commit2act
# [TIG-662] Route names for signup flow

## Ticket

* [TIG-662](https://app.clickup.com/t/9009201449/TIG-662)

## Description

This PR adds sub-routes for the updated Sign Up flow via the `SIGNUP_ROUTES` array. From there, each route is added to a new `PublicRoutes` component which alternately renders the existing `Login` page or the Sign Up pages, depending on the current route. The `SignupContainer` component introduced here is intentionally bare-bones (as described in the ticket scope), rendering only a static heading and Back/Next links. 

## Testing

* Pull down and install the branch
* Run local storybook using `npm run storybook`
* Confirm the Views > `SignupContainer` story renders and the `step` prop can be updated using the Controls panel
* Shut down storybook and run tests using `npm run test`
* Confirm tests for `SignupContainer`, `PublicRoutes`, and `App` all pass
* Shut down the tests and run the local app using `npm run start`
* With no active user session, confirm the Login page renders for the home (`http://localhost:3000/`) route
* Select "Create an Account" and confirm the existing sign up panel renders
* Navigate to `http://localhost:3000/account/create` and confirm that placeholder page appears
* Use the Next / Back links to jump between Sign Up flow pages, confirming the URL updates with each

## Screenshots

**First step in URL, "back" link renders static text:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/01528280-8ea8-4286-bc44-1e787c813642)

**Later step in URL, both "back" and "next" links rendering:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/4cb832e6-3b1d-4230-9e20-2cac5b782969)

**Last step in URL, "next" link renders static text:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/48269cd4-d2c3-48b7-ba8c-899e24dcb887)

**Stories for `SignupContainer` in Storybook, with adjustable `step` value in the Controls panel:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/997a0bf5-ec4f-4558-8f84-12e8625a9f67)
